src/server/page_server.hpp

```diff
@@ -182,7 +182,56 @@ class page_server
     };
 
     /*
-     *  TODO add some explanation and diagrams for this.
+     *  When a disconnected page server re-connects to an ATS, or a cluster, the cluster possibly
+     *  has gone further than the page server. It means the connecting PS has log records only up to LSA(N),
+     *  but the ATS and other PSes in the cluster have log records up to a LSN larger than LSN(N) like LSN(N+1000).
+     *  Because ATS sends log records and just discards them, the connecting PS must get the pages from another PS to fill the hole.
+     *
+     *  Terms:
+     *  - catchup_lsa: the log lsa where new log records from ATS start to come in. A PS must have log records
+     *    until this point. If it doesn’t have, it must fetch log pages until the point by the catch-up mechanism.
+     *    Note that the catchup_lsa is different depending on when a PS connects to an ATS.
+     *  - Follower: a page server who is reconnecting to ATS with log records in the past. It must fill the log records hole
```

Comment:
Nit.
"reconnecting to ATS"
IMHO, a PS does not reconnect to the ATS specifically; it re-attaches/re-connects to a running cluster.
We all know what this means. My remark regards what the comment conveys to future/external readers. :)

Reply:
done!